From 2e8632ae0ac799d4ad7cf253530576c53369baa7 Mon Sep 17 00:00:00 2001
From: Chris Liddell <chris.liddell@artifex.com>
Date: Fri, 8 Jan 2016 11:00:08 +0000
Subject: [PATCH] Bug 696498: clean up endian configure checks

The custom endian checks in the configure scripts were implemented because the
built-in test in the most common autoconf version at the time was buggy on
several platforms.

Those autoconf problems were addressed some time and several versions ago, and
the newer versions are widely adopted, so we can safely use the built-in
AC_C_BIGENDIAN() test.
---
 configure.ac | 31 ++-----------------------------
 1 file changed, 2 insertions(+), 29 deletions(-)

diff --git a/configure.ac b/configure.ac
index f533e464a80f..9ca792f37509 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,7 @@ dnl Initialization and Versioning
 dnl ------------------------------------------------
 
 AC_INIT
-AC_PREREQ(2.60)
+AC_PREREQ(2.63)
 AC_LANG(C)
 AC_CONFIG_SRCDIR(psi/gs.c)
 
@@ -262,34 +262,7 @@ dnl ----------------------------
 dnl check for big/little endian
 dnl ----------------------------
 
-BIGENDIAN=
-
-AC_MSG_CHECKING([for big endian])
-
-AC_ARG_ENABLE([big-endian], AC_HELP_STRING([--enable-big-endian],
-    [Force big endian]), [BIGENDIAN=1])
-
-if test x$BIGENDIAN = x; then
-  AC_ARG_ENABLE([little-endian], AC_HELP_STRING([--enable-little-endian],
-      [Force little endian]), [BIGENDIAN=0])
-fi
-
-if test x$BIGENDIAN = x ; then
-  AC_RUN_IFELSE(
-    [AC_LANG_PROGRAM([#include <stdio.h>], [
-     static const int one = 1;
-     return (*(char*)&one == 0 ? 0 : 1);
-    ])],
-    [BIGENDIAN=1;AC_MSG_RESULT(big)],
-    [BIGENDIAN=0;AC_MSG_RESULT(little)],
-    [BIGENDIAN=1;AC_MSG_RESULT(cross compiling: assuming big engian, otherwise use "--enable-little-endian")])
-else
-  if test x$BIGENDIAN = x1 ; then
-    AC_MSG_RESULT(big)
-  else
-    AC_MSG_RESULT(little)
-  fi
-fi
+AC_C_BIGENDIAN([BIGENDIAN=1], [BIGENDIAN=0])
 
 dnl --------------------------------------------------
 dnl check for sse2 intrinsics
-- 
2.8.0.rc3.226.g39d4020

